### PR TITLE
implement redaction filters

### PR DESCRIPTION
**Pull request by willfindlay**
Implement a new field filter type, the redaction filter. Redaction filters use regular expressions to suppress sensitive information in string fields in Tetragon events. When a regular expression in a redaction filter matches a string, everything inside of its capture groups is replaced with `*****`, effectively censoring the output. For example, the regular expression `(?:--password|-p)(?:\s+|=)(\S*)` will convert the string "--password=foo" into "--password=*****".

In some cases, it is not desirable to apply a redaction filter to all events. For this use case, redaction filters also include an event filter which can be used to select events to redact. This event filter is configured with the same syntax as an export filter. As a more concrete example:

    {"match": {"binary_regex": ["^foo$"]}, "redact": ["\W(qux)\W"]}

The above filter would redact any occurrences of the word "qux" in events with the binary name "foo".

Due to the sensitive nature of redaction, these filters are applied as configured in the
agent, regardless of whether an event is exported via gRPC or the JSON exporter. In other
words, redaction filter configuration always happens at the agent config level, not in the
gRPC client CLI.

```release-note
Introduce redaction filters for censoring sensitive string data in process events.
```

Fixes: #2241

### CODE REVIEW

**Comment by willfindlay** [comment]
marking as draft while I fix an oversight with empty redaction filters

File: cmd/tetragon/main.go [hunk 1]
```diff
@@ -401,11 +406,18 @@ func tetragonExecute() error {
 
 	hookRunner := rthooks.GlobalRunner().WithWatcher(k8sWatcher)
 
+	redactionFilters, err := getRedactionFilters()
+	if err != nil {
+		return err
+	}
+	log.WithFields(logrus.Fields{"redactionFilters": redactionFilters}).Debug("Configured redaction filters")
```
Comment (kkourt) [inline cmd/tetragon/main.go:413]: IMO, I think `Info` is better  here.

File: install/kubernetes/tetragon/values.yaml [hunk 2]
```diff
@@ -111,7 +111,33 @@ tetragon:
...
+  # Now, an event which contains the string "--password=foo" would have that
+  # string replaced with "--password=*****".
+  # 
+  # Suppose we also see some passwords passed via the -p shorthand for a specific binary, foo.
+  # We can also redact these as follows:
+  #
+  #   {"match": [{"binary_regex": "(?:^|/)foo$"}], "redact": ["-p(?:\s+|=)(\S*)"]}
+  #
+  # With both of the above redaction filters in place, we are now redacting all
+  # password arguments.
```
Comment (kkourt) [inline install/kubernetes/tetragon/values.yaml:138]: ❤️

**Review by kkourt — APPROVED** [review]
LGTM, thanks!

File: cmd/tetragon/main.go (hunk 1 above)
Comment (willfindlay) [reply cmd/tetragon/main.go:413]: Sure why not. For the record we're also dumping other filter types in debug logs. Maybe worth changing those too

Comment (willfindlay) [reply cmd/tetragon/main.go]: Updated

**Review by jrfastab — APPROVED** [review]

**Pull request merged**